### ISSUE TITLE
Dirty and modified states are not correct in Wizard form

### ISSUE THREAD

**Issue by sedx** [final-form/final-form#288]
### Are you submitting a **bug report** or a **feature request**?
bug report
### What is the current behavior?
1) Change fisrt name
2) Click Next
3) Check the dirty state and modified fields
4) Pick color
5) Click Next
6) Check the dirty state and modified fields

### What is the expected behavior?

* On step 3 dirty state of form should be true, because initial values are not match to current values and modified fields should contain `firstName` key
* On step 6 dirty state should be true too and modified fields should contain `firstName` and `favoriteColor` keys (because they are was modified)

### Sandbox Link

https://codesandbox.io/s/wizard-example-cpset?fontsize=14

### What's your environment?

react-final-form 4.18.5

### Other information

In my project I have complicated form, that split to multiple tabs (so form looks like wizard).
But user can save from data from any step, not from the last. Save button enabled only when form is in dirty state.
Tabs are lazy loaded and when we change tab current `<Field />` components are unmounted.

When I change some values in form on tab A - form becomes dirty, when I goes to tab B changed fields are unmounted, changed values still stored in value (it is ok), but dirty state now are false (it's looks like not ok).

Actually form is dirty, because my initial value is not match the current form value.
Also list of modified fields are changed too (it's drops fields from A's tab). It's not represent actual diff from initial and current value.

**Comment by sedx** [comment]
Yes, as I told, when field unmount new form's state calculated and forms global dirty and modified based on list of current registered fields
For dirty:
https://github.com/final-form/final-form/blob/a95948ab8f2804479b5a6620b88b48525472f0db/src/FinalForm.js#L516-L542 

For  `modified`, `touched` and `visited`:
https://github.com/final-form/final-form/blob/a95948ab8f2804479b5a6620b88b48525472f0db/src/FinalForm.js#L554-L562
I can create MR to keep global dirty state and list of modified unmounted fields.

There is two way:
1) Fix it and reflect actual form state out of box (look like breaking changes, or may be reason for unexpected behaviour in other projects)
2) Add `keepFormStateOnFieldUnmount` to config and dependent on values of this attribute keep current behaviour, or keep actual form state.

@erikras What do you think?

**Comment by sedx** [comment]
I think for final-form config name should be `keepFormStateOnFieldUnregister` (there is not 'unmount' for fields in final-form)

`keepFormStateOnFieldUnmount` will be props for react-final-form (where is fields unmounted)

**Comment by BLRplex** [comment]
As I understand nobody cares about this issue?

**Comment by aleshanter** [comment]
@sedx do you have any updates on this issue?

**Comment by sedx** [comment]
@aleshanter  No 😅 I have no answer about which way to choose for fix, so I did not start with changes. On project I was check initial values and current state for diff (I supose)

I checked codesandbox with latest version - issue is still here.
